logger.go

```diff
@@ -0,0 +1,27 @@
+package nebula
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+type ContextualError struct {
```

Comment:
This should also implement the `Unwrap() error` method expected on error wrapping types.